packages/babel-register/src/experimental-worker.js

```diff
@@ -13,7 +13,11 @@ if (major < 12 || (major === 12 && minor < 3)) {
 const hook = require("./hook");
 const { WorkerClient } = require("./worker-client");
 
-const register = hook.register.bind(null, new WorkerClient());
```

Comment:
This was spawning infinite workers when using `--require` in `NODE_OPTIONS`, becuse it was not guarded by the `if (!require("./is-in-register-worker").isInRegisterWorker)` check below.